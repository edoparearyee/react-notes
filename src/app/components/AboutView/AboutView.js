import React, { Component } from 'react';
import { PropTypes } from 'prop-types';

import './AboutView.scss';

class AboutView extends Component {
  componentDidMount() {
    this.props.fetchContributors();
  }

  componentWillUnmount() {
    this.props.clearContributors();
  }

  render() {
    const { error, loading, items } = this.props;
    let contributors;

    if (error) {
      contributors = <p>Error! {error.message}</p>;
    } else if (loading) {
      contributors = <p>Loadingâ€¦</p>;
    } else {
      contributors = (
        <ul className="ContributorList">
          {items.map(item => (
            <li key={item.id} className="ContributorList__Item">
              <img
                className="ContributorList__Image"
                src={item.avatar_url}
                alt={item.login}
              />
              <p className="Contributor__Name">{item.login}</p>
            </li>
          ))}
        </ul>
      );
    }

    return (
      <div className="AboutView">
        <h1>About</h1>
        <p>
          Officia cillum ea incididunt laborum velit. Deserunt voluptate
          exercitation deserunt ex occaecat fugiat enim incididunt sit est.
          Exercitation laborum ex fugiat in amet laboris reprehenderit velit
          tempor dolore nostrud do.
        </p>
        <h2>Contributors</h2>
        {contributors}
      </div>
    );
  }
}

AboutView.propTypes = {
  loading: PropTypes.bool.isRequired,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      login: PropTypes.string,
      id: PropTypes.number,
      node_id: PropTypes.string,
      avatar_url: PropTypes.string,
      gravatar_id: PropTypes.string,
      url: PropTypes.string,
      html_url: PropTypes.string,
      followers_url: PropTypes.string,
      following_url: PropTypes.string,
      gists_url: PropTypes.string,
      starred_url: PropTypes.string,
      subscriptions_url: PropTypes.string,
      organizations_url: PropTypes.string,
      repos_url: PropTypes.string,
      events_url: PropTypes.string,
      received_events_url: PropTypes.string,
      type: PropTypes.string,
      site_admin: PropTypes.bool,
      contributors: PropTypes.number,
    }),
  ).isRequired,
  error: PropTypes.object,
  fetchContributors: PropTypes.func.isRequired,
  clearContributors: PropTypes.func.isRequired,
};

export default AboutView;
