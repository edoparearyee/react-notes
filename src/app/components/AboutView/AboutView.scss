.AboutView {
  display: block;
}

.ContributorList__Item {
  align-items: center;
  display: flex;
}

.ContributorList__Image {
  height: 40px;
  margin-right: 5px;
  width: 40px;
}
