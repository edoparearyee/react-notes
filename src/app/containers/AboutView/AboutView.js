import { connect } from 'react-redux';

import AboutView from '../../components/AboutView/AboutView';
import {
  fetchContributors,
  contributorsClear,
} from '../../actions/contributors';

const mapStateToProps = state => {
  return {
    loading: state.contributors.loading,
    items: state.contributors.items,
    error: state.contributors.error,
  };
};

const mapDispatchToProps = dispatch => ({
  fetchContributors: () => fetchContributors(dispatch),
  clearContributors: () => dispatch(contributorsClear()),
});

const AboutViewContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(AboutView);

export default AboutViewContainer;
