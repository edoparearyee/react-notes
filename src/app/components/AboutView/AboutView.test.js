import React from 'react';
import ReactDOM from 'react-dom';
import AboutView from './AboutView';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AboutView />, div);
  ReactDOM.unmountComponentAtNode(div);
});
