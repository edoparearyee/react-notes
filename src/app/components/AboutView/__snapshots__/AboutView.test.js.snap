// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView renders when loading 1`] = `
<div
  className="AboutView"
>
  <h1>
    About
  </h1>
  <p>
    Officia cillum ea incididunt laborum velit. Deserunt voluptate exercitation deserunt ex occaecat fugiat enim incididunt sit est. Exercitation laborum ex fugiat in amet laboris reprehenderit velit tempor dolore nostrud do.
  </p>
  <h2>
    Contributors
  </h2>
  <p
    className="LoadingMessage"
  >
    Loadingâ€¦
  </p>
</div>
`;

exports[`AboutView renders when not loading 1`] = `
<div
  className="AboutView"
>
  <h1>
    About
  </h1>
  <p>
    Officia cillum ea incididunt laborum velit. Deserunt voluptate exercitation deserunt ex occaecat fugiat enim incididunt sit est. Exercitation laborum ex fugiat in amet laboris reprehenderit velit tempor dolore nostrud do.
  </p>
  <h2>
    Contributors
  </h2>
  <ul
    className="ContributorList"
  >
    <li
      className="ContributorList__Item"
      key="0"
    >
      <img
        alt="foo"
        className="ContributorList__Image"
        src="http://foo.com/image1.jpg"
      />
      <p
        className="Contributor__Name"
      >
        foo
      </p>
    </li>
    <li
      className="ContributorList__Item"
      key="1"
    >
      <img
        alt="bar"
        className="ContributorList__Image"
        src="http://foo.com/image2.jpg"
      />
      <p
        className="Contributor__Name"
      >
        bar
      </p>
    </li>
  </ul>
</div>
`;

exports[`AboutView renders when there is an error 1`] = `
<div
  className="AboutView"
>
  <h1>
    About
  </h1>
  <p>
    Officia cillum ea incididunt laborum velit. Deserunt voluptate exercitation deserunt ex occaecat fugiat enim incididunt sit est. Exercitation laborum ex fugiat in amet laboris reprehenderit velit tempor dolore nostrud do.
  </p>
  <h2>
    Contributors
  </h2>
  <p
    className="ErrorMessage"
  >
    Error! 
    Something went wrong!
  </p>
</div>
`;
